File: 백준/Silver/1783. 병든 나이트/병든 나이트.java
import java.util.*;

public class Main {
    
    public static void main(String args[])
    {
      Scanner s = new Scanner(System.in);
      int n = s.nextInt(),m= s.nextInt();
      if(n==1) System.out.print(1);
      else if(n==2)System.out.print(Math.min(4,(m+1)/2));
      else if(n>=3)
      {
          if(m<=6)System.out.print(Math.min(4,m));
          else System.out.print(m-2);
      }
    }
}

File: 백준/Silver/1783. 병든 나이트/병든 나이트.kt
import java.util.*

fun main() {
    val s = Scanner(System.`in`)
    var n = s.nextInt();var m = s.nextInt()
    
    if(n==1) print(1)
    else if(n==2) print(Math.min(4,(m+1)/2))
    else if(n>=3)
    {
        if(m<=6)print(Math.min(4,m))
        else print(m-2)
    }
}